helix-view/src/editor.rs

```diff
@@ -94,15 +103,22 @@ impl Editor {
         themes: Arc<theme::Loader>,
         config_loader: Arc<syntax::Loader>,
         config: Config,
-    ) -> Self {
+    ) -> anyhow::Result<Self> {
         let language_servers = helix_lsp::Registry::new();
+        let (watcher, watcher_receiver) = {
+            let (watcher_sender, watcher_receiver) = unbounded_channel();
+            let watcher = NotifyActor::spawn(Box::new(move |e| watcher_sender.send(e).unwrap()))
+                .context("Failed to spawn watcher")?;
+            (Arc::new(watcher), watcher_receiver)
+        };
```

Comment:
Unless I'm misunderstanding something, it looks like this API makes it impossible to open an editor unless spawning the file watcher is successful.  Presumably, though, we still want to succeed in opening the editor, and just report to the user that file watching isn't active, right?  For example, I still want Helix to work even if I'm on a system that doesn't provide the expected notify capabilities.